lib/segment/src/index/hnsw_index/search_context.rs

```diff
@@ -24,19 +22,13 @@ impl SearchContext {
     }
 
     pub fn lower_bound(&self) -> ScoreType {
-        match self.nearest.top() {
-            None => ScoreType::min_value(),
-            Some(worst_of_the_best) => worst_of_the_best.score,
-        }
+        self.nearest.threshold()
```

Comment:
Why is it Ok. to use threshold here given that it is lazy and might not represent the full information of the top values?

Reply:
This might be not Ok, it is up to benchmarks